### PR TITLE
New Template for Cisco NXOS - show_interface_transceiver

### PR DESCRIPTION
**Pull request by dgarros**
##### ISSUE TYPE
 - New Template Pull Request

##### COMPONENT
Template: cisco_nxos_show_interface_transceiver.textfsm
OS: Cisco NXOS
Command: `show interface transceiver`

##### SUMMARY
Extract transceiver information (serial number, type, part, name) from `show interface transceiver` on Cisco NXOS

### CODE REVIEW

File: templates/index [hunk 1]
```diff
@@ -202,6 +202,7 @@ cisco_ios_dir.textfsm,  .*, cisco_ios, dir
 cisco_nxos_show_l2rib_internal_permanently-frozen-list.textfsm, .*, cisco_nxos, sh[[ow]] l2ri[[b]] i[[nternal]] pe[[rmanently-frozen-list]]
 cisco_nxos_show_configuration_session_summary.textfsm, .*, cisco_nxos, sh[[ow]] configu[[ration]] s[[ession]] su[[mmary]]
 cisco_nxos_show_interface_transceiver_details.textfsm, .*, cisco_nxos, sh[[ow]] int[[erface]] tra[[nsceiver]] de[[tails]]
+cisco_nxos_show_interface_transceiver.textfsm, .*, cisco_nxos, sh[[ow]] int[[erface]] tra[[nsceiver]]
```
Comment (FragmentedPacket) [inline templates/index]: Should be ordered in length and is a bit too high up in the ordering. I think it's only a few lines off.

If you have **tox** installed, you should be able to run **tox** from the root directly and it'll build the necessary environment and run the tests.

File: templates/cisco_nxos_show_interface_transceiver.textfsm [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+Value PART_NUMBER (\S+)
+
+Start
+  ^${INTERFACE}
+  ^\s+type\s+is\s+${TYPE}
+  ^\s+name\s+is\s+${MANUFACTURER} 
+  ^\s+part\s+number\s+is\s+${PART_NUMBER}
+  ^\s+serial\s+number\s+is\s+${SERIAL} -> Record
+
+EOF
```
Comment (FragmentedPacket) [inline templates/cisco_nxos_show_interface_transceiver.textfsm]: You can remove the EOF and then if you don't mind adding the following to the end of the **Start** state:
```
  ^\s*$$
  ^. -> Error
```
Once you make the change, you'll most likely have to account for the lines that aren't currently being captured. You can typically just add the first word or two of a line to account for it.

Comment (dgarros) [reply templates/cisco_nxos_show_interface_transceiver.textfsm]: @FragmentedPacket thanks for review this PR
I made the changes as suggested

File: templates/cisco_nxos_show_interface_transceiver.textfsm [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Value INTERFACE (\S+/\S+)
+Value MANUFACTURER (\S+)
+Value TYPE (.+)
+Value SERIAL (\S+)
+Value PART_NUMBER (\S+)
+
+Start
+  ^${INTERFACE}
```
Comment (jmcgill298) [inline templates/cisco_nxos_show_interface_transceiver.textfsm]:
```suggestion
  ^\S+/\S+ -> Continue.Record
  ^${INTERFACE}
```

File: templates/cisco_nxos_show_interface_transceiver.textfsm [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  ^\s+type\s+is\s+${TYPE}
+  ^\s+name\s+is\s+${MANUFACTURER} 
+  ^\s+part\s+number\s+is\s+${PART_NUMBER}
+  ^\s+serial\s+number\s+is\s+${SERIAL} -> Record
```
Comment (jmcgill298) [inline templates/cisco_nxos_show_interface_transceiver.textfsm]:
```suggestion
  ^\s+serial\s+number\s+is\s+${SERIAL}
```

File: templates/cisco_nxos_show_interface_transceiver.textfsm [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  ^\s+transceiver\s+
+  ^\s+nominal\s+
+  ^\s+revision\s+
+  ^\s+Link\s+
+  ^\s+cisco\s+id
+  ^\s+cisco\s+extended
+  ^\s*$$
+  ^. -> Error
```
Comment (jmcgill298) [inline templates/cisco_nxos_show_interface_transceiver.textfsm]:
```suggestion
  ^. -> Error
  
```

Comment (dgarros) [reply templates/cisco_nxos_show_interface_transceiver.textfsm]: Thanks @jmcgill298 I made the changes as suggested

**Comment by jmcgill298** [comment]
Thanks @dgarros

**Pull request merged**